includes/handlers/class-webmention-handler-mf2.php

```diff
@@ -437,7 +437,7 @@ protected function get_location( $mf ) {
 	 * @param array Author array.
 	 */
 	protected function get_author( $mf_array ) {
-		$properties = $this->get_property( $mf_array, 'author' );
+		$properties = current( $this->get_property( $mf_array, 'author' ) );
```

Comment:
this throws an error if `get_property` does not return an array. try `https://notiz.blog`.

Reply:
Shouldn't it always be an array in mf2 json?

Reply:
maybe, but in this case it isn't. maybe, because `notiz.blog` is an `h-feed` and not an `h-entry`.

Reply:
the representative function is supposed to extract the correct h-entry

Reply:
I see the problem. It's looking for a representative h-entry...but notiz.blog would be an author/feed page. When would an h-card mention another site?

Reply:
???

Reply:
You said you used it on notiz.blog. When would that be the source of a webmention?

Reply:
Maybe never, but we can not predict the usage of other sites, so the parser should be able to handle such cases. Or one step further the parser should never return php errors or warnings, regardless of the site.

Reply:
The "legacy" wordpress themes for example are using hfeed on every page...

Reply:
Yes, so the code extracts the h-entry whose u-url matches the source url.

Reply:
Then it does author discovery and sets it to the author property if there isn't one.

Reply:
It was meant to give it a 'clean' object to work off of and save.